mesonbuild/optinterpreter.py

```diff
@@ -77,9 +77,6 @@ def __init__(self, subproject, command_line_options):
             if self.subproject != '': # Strip the beginning.
                 if not o.startswith(self.sbprefix):
                     continue
-            else:
-                if ':' in o:
-                    continue
```

Comment:
We're permitting it in option values, not in option keys. That's what the original bug was about. We cannot have an option key with a colon in it since it must match `optname_regex` which is `[^a-zA-Z0-9_-]`.